### PR TITLE
feat: add new CronTrigger task type

### PR DESCRIPTION
**Pull request by zevaryx**
## Pull Request Type

- [x] Feature addition
- [ ] Bugfix
- [ ] Documentation update
- [ ] Code refactor
- [ ] Tests improvement
- [ ] CI/CD pipeline enhancement
- [ ] Other: [Replace with a description]

## Description

Adds a new Task Trigger, CronTrigger. Pass a cron string to it (i.e. `0 0 * * *`), and it'll run tasks based on that cron string

## Changes

- Add `croniter` requirement
- Add `CronTrigger` task trigger

## Test Scenarios

```py
@Task.create(CronTrigger("* * * * *"))
async def test_cron(self) -> None:
    print("I run every minute!")
```

## Python Compatibility

- [x] I've ensured my code works on Python `3.10.x`
- [x] I've ensured my code works on Python `3.11.x`

## Checklist

- [ ] I've run the `pre-commit` code linter over all edited files
- [x] I've tested my changes on supported Python versions
- [x] I've added tests for my code, if applicable
- [x] I've updated / added  documentation, where applicable

### CODE REVIEW

**Comment by AstreaTSS** [comment]
This debatably solves #1560.

**Review by silasary — APPROVED** [review]

**Comment by silasary** [comment]
> This debatably solves https://github.com/interactions-py/interactions.py/issues/1560.

I think it does, yeah

File: interactions/models/internal/tasks/triggers.py [hunk 1]
```diff
@@ -140,3 +142,20 @@ def _set_current_trigger(self) -> BaseTrigger | None:
...
         return self.current_trigger.next_fire() if self._set_current_trigger() else None
+
+
+class CronTrigger(BaseTrigger):
+    """
+    Trigger the task based on a cron schedule.
+
+    Attributes:
+        cron str: The cron schedule, use https://crontab.guru for help
+        utc bool: Whether or not to use UTC for the time
```
Comment (AstreaTSS) [inline interactions/models/internal/tasks/triggers.py]: Seems like you forgot to change this part 😅

**Review by AstreaTSS — CHANGES_REQUESTED** [review]

File: interactions/models/internal/tasks/triggers.py (hunk 1 above)
Comment (eightween) [reply interactions/models/internal/tasks/triggers.py]: What about this needs to be changed?

Comment (AstreaTSS) [reply interactions/models/internal/tasks/triggers.py]: The `utc` docstring entry needs to be replaced with a `tz` entry.

**Review by AstreaTSS — APPROVED** [review]
Don't know what's going on with the PR's diff, but looks fine otherwise.

**Pull request merged**